subprojects/core/src/integTest/groovy/org/gradle/api/internal/changedetection/state/TaskTypeUpToDateIntegrationTest.groovy

```diff
@@ -176,6 +180,33 @@ class TaskTypeUpToDateIntegrationTest extends AbstractIntegrationSpec {
         then: skipped(":copy")
     }
 
+    @Issue("https://github.com/gradle/gradle/issues/9723")
+    def "declaring a task action receiving #incrementalChangesType without declaring outputs is deprecated"() {
+        def input = file('input.txt').createFile()
+        buildFile << """
+            class Task extends DefaultTask {
```

Comment:
we could maybe call it something else, like `IncrementalTask`

Reply:
Done.